### ISSUE TITLE
OOM with A100 on Mistral 7B LoRA fine tuning (load in 8bit)

### ISSUE THREAD

**Issue by Palmik** [axolotl-ai-cloud/axolotl#814]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Fine tuning Mistral 7B LoRA (load_in_8bit = True, sequence_len=4096, micro_batch_size=2) goes OOM, reducing to micro_batch_size=1 works.

After changing to micro_batch_size, I get this in the logs:

```
[2023-11-02 18:56:22,389] [INFO] [axolotl.load_model:404] [PID:5470] [RANK:0] GPU memory usage after model load: 7.494GB (+0.114GB cache, +1.341GB misc)
```

And then later during training:

```
[2023-11-02 18:57:01,107] [INFO] [axolotl.callbacks.on_step_end:122] [PID:5470] [RANK:0] GPU memory usage while training: 13.270GB (+56.066GB cache, +1.374GB misc)
```

Is the cache expected to take so much?

### Current behaviour

Currently OOMs.

### Steps to reproduce

Use the provided axolotl config on A100 80G

### Config yaml

```
base_model: mistralai/Mistral-7B-v0.1
model_type: MistralForCausalLM
tokenizer_type: LlamaTokenizer
is_mistral_derived_model: true

load_in_8bit: true
load_in_4bit: false
strict: false

datasets:
  - path: /workspace/merged.json
    type: completion
dataset_prepared_path: last_run_prepared
val_set_size: 0
output_dir: ./mistral-7b-lora-out

adapter: lora
lora_model_dir:

sequence_len: 4096
sample_packing: true
pad_to_sequence_len: true

lora_r: 64
lora_alpha: 16
lora_dropout: 0.05
lora_target_linear: true
lora_fan_in_fan_out:
lora_target_modules:
  - gate_proj
  - down_proj
  - up_proj
  - q_proj
  - v_proj
  - k_proj
  - o_proj

wandb_project:
wandb_entity:
wandb_watch:
wandb_run_id:
wandb_log_model:

gradient_accumulation_steps: 4
micro_batch_size: 2
num_epochs: 2
optimizer: adamw_bnb_8bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: true
group_by_length: false
bf16: true
fp16: false
tf32: false

gradient_checkpointing: false
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

warmup_steps: 10
# eval_steps: 20
# eval_table_size:
# eval_table_max_new_tokens: 128
save_steps:
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  bos_token: "<s>"
  eos_token: "</s>"
  unk_token: "<unk>"
```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

Using official docker image, see below

### axolotl branch-commit

Using main-latest https://hub.docker.com/layers/winglian/axolotl-runpod/main-latest/images/sha256-a5910803b3c1c4286f4b901966a1aa5ca3964cd231a920635a4ca3e4ee5570d5

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by Palmik** [comment]
I updated to the new docker image from few hours ago + also enabled gradient checkpointing and now A100 can do microbatch size 8. It could be due to gradient checkpointing, though I would not have expected 8x difference with 8bit adam.
